modules/fluid_properties/src/userobjects/HeliumFluidProperties.C

```diff
@@ -30,6 +30,38 @@ HeliumFluidProperties::fluidName() const
   return "helium";
 }
 
+ADReal
+HeliumFluidProperties::e_from_p_rho(const ADReal & p, const ADReal & rho) const
+{
+  // Initial guess using ideal gas law
+  ADReal e = p / (_cp / _cv - 1.) / rho;
+  const ADReal v = 1. / rho;
+
+  Real real_p_from_props, dp_dv, dp_de;
+  const unsigned int max_its = 10;
+  unsigned int it = 0;
+
+  do
+  {
+    p_from_v_e(v.value(), e.value(), real_p_from_props, dp_dv, dp_de);
+    ADReal p_from_props = real_p_from_props;
```

Comment:
`residual, Jacobian, etc.` are part of forming the Newton update `delta_e` which is used directly in computing `e` itself. So having everything be `ADReal` is necessary for getting the `derivatives` in `e` correct.

However, I'm wary of this method for another reason... this will not capture the fact that `dp_dv` and `dp_de` are themselves functions of the nonlinear degrees of freedom unless `p` is a linear function of `v` or `e`.

I may actually remove this override for that reason

Reply:
I did not remove this override but I added the correct nonlinear dof dependence of `dp_dv` and `dp_de`.